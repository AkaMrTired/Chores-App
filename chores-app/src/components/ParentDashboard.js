import React, { useEffect } from "react";
import axios from "axios";
import ChoreCard from "./ChoreCard";
import "../styles/ParentDashboard.css";

import { useUserAuth } from "../context/UserAuthContext";

// eslint-disable-next-line react/prop-types
const ParentDashboard = () => {
  const { chores, setChores } = useUserAuth();
  useEffect(() => {
    const familyID = localStorage.getItem("familyID");
    if (familyID) {
      axios
        .get(`http://localhost:3300/family/${familyID}/chores`)
        .then((response) => {
          setChores(response.data);
        })
        .catch((e) => {
          console.log(e);
        });
    } else {
      console.log("no family ID");
    }
  }, []);
  return (
    <div className="container parent-dashboard-container">
      <h1>Dashboard</h1>
      <h2>My Family</h2>
      <button type="button" className="btn btn-narrow btn-stroke_white">
        <a href="/newmember">+ Invite new member</a>
      </button>
      <button type="button" className="btn btn-narrow btn-fill_purple">
        <a href="/listofchildren">View Children</a>
      </button>
      <h2>Chores</h2>
      <div className="btn-container">
        <button type="button" className="btn btn-narrow btn-stroke_white">
          <a href="/addchore">+ Add new chore</a>
        </button>
        <button type="button" className="btn btn-fill_peach">
          <a href="/approvechores">Manage chores pending approval</a>
        </button>
      </div>
      {/* mapping function to go through the chores list and render them. */}
      {chores.map((chore) => (
        <ChoreCard
          key={chore.choreID}
          name={chore.name}
          price={chore.price}
          status={chore.status}
          choreID={chore.choreID}
          choreOwner={chore.owner}
          component="ParentDashboard"
        />
      ))}
      <div />
    </div>
  );
};

export default ParentDashboard;
