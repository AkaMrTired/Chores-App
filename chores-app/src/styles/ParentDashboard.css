.parent-dashboard-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.parent-dashboard-chores-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 4rem;
  margin-bottom: 2rem;
  background: var(--gradient);
  padding: 0.5rem;
  border-radius: 25px;
}

.parent-dashboard-chores-container h2 {
  color: var(--purple-dark);
  text-align: center;
}

.parent-dashboard-chores-container .btn-narrow {
  margin-bottom: 1.5rem;
}
