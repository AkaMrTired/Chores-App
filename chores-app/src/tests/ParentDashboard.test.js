import { render, screen } from "@testing-library/react";
import React from "react";
import ParentDashboard from "../components/ParentDashboard";

const validChores = [];

describe("the parents dashboard", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<ParentDashboard chores={validChores} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<ParentDashboard chores={validChores} />);
    expect(screen.getByText(/dashboard/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /invite new member/i })
    ).toBeInTheDocument();
    expect(screen.getByText("Chores")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /add new chore/i })
    ).toBeInTheDocument();
  });
});
