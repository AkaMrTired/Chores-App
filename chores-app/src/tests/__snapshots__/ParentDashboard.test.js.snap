// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the parents dashboard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Dashboard
    </h1>
    <button
      type="button"
    >
      <a
        href="/newmember"
      >
        + Invite new member
      </a>
    </button>
    <div
      class="container"
    >
      <h1>
        Chores
      </h1>
      <button
        type="button"
      >
        <a
          href="/addchore"
        >
          Add new chore +
        </a>
      </button>
    </div>
    <div>
      <button
        type="button"
      >
        <a
          href="/approvechores"
        >
          Manage chores pending approval
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
